app/utils/utils.ts

```diff
@@ -46,3 +47,86 @@ export const dollarFormat = (value: number) => {
+ * @param validationType
+ * @param dataType
+ * @returns Record<string, any>
+ */
+
+// Sample reusable function to get field validation parameters
+export const getFieldValidation = (validationCriteria: string, validationType: string, dataType: string) => {
+  const validationRules: Record<string, any> = {};
+  if (dataType === "true-false") {
+    validationRules["type"] = "true-false";
+    return validationRules;
+  }
+
+  switch (validationType) {
+    case "=": // Data equal to
+      validationRules["equals"] = validationCriteria;
+      break;
+
+    case "regex": // Text Pattern (regex)
+      validationRules["pattern"] = new RegExp(validationCriteria);
+      break;
+
+    case ">=": // Number greater than or equal to
+      validationRules["min"] = dataType === "number-input" ? Number(validationCriteria) : dayjs(validationCriteria);
+      break;
+
+    case ">": // Number greater than
+      validationRules["min"] = dataType === "number-input" ? Number(validationCriteria) + 1 : dayjs(validationCriteria);
+      break;
+
+    case "<=": // Number less than or equal to
+      validationRules["max"] = dataType === "number-input" ? Number(validationCriteria) : dayjs(validationCriteria);
+      break;
+
+    case "<": // Number less than
+      validationRules["max"] = dataType === "number-input" ? Number(validationCriteria) - 1 : dayjs(validationCriteria);
+      break;
+
+    case "(num)": // Number within range (exclusive)
+    case "[num]": // Number within range (inclusive)
+    case "(date)": // Date within range (exclusive)
+    case "[date]": // Date within range (inclusive)
+      const range = validationCriteria.replace(/[\[\]\(\)]/g, "").split(",");
+      if (range.length === 2) {
+        validationRules["range"] = {
+          min: dataType === "number-input" ? Number(range[0].trim()) : dayjs(range[0].trim()),
+          max: dataType === "number-input" ? Number(range[1].trim()) : dayjs(range[1].trim()),
+          inclusive: validationType.startsWith("["), // true for inclusive ranges
+        };
+      }
+      break;
+
+    case "[=text]": // Text Options
+    case "[=num]": // Number Options
+    case "[=date]": // Date Options
+      const options: { value: string | number | Date | null; type: string | null | undefined; label?: string }[] =
+        validationCriteria
+          .replace(/[\[\]]/g, "")
+          .split(",")
+          .map((value) => ({ value: value, type: typeof value }));
+      options.push({ value: null, type: null, label: "No Value" }); // Add null as the last option
+      validationRules["options"] = options;
+      break;
+
+    default:
+      console.warn(`Unknown validation type: ${validationType}`);
+  }
+
+  // Enforce proper type-based validation
+  if (dataType === "number-input") {
+    validationRules["type"] = "number";
+  } else if (dataType === "date") {
+    validationRules["type"] = "date";
+  } else {
+    validationRules["type"] = "text";
+  }
+
+  return validationRules;
+};
```

Comment:
Updated some to better match the Zen Expression Language (= to ==, and introducing !=). The majority are similar or identical, but my intent with klamm was to have more specific naming conventions for ease of understanding. Everything should be handled identically besides ranges, as I didn't want to over complicate it. We use the same () and [] brackets to signify exclusive/inclusive ranges, but because we deal with a lot of float numbers i didn't want to use this format from zen:
[0..10]      [// true, (internally this is $ >= 0 and $ <= 10) ](https://gorules.io/docs/user-manual/decision-modeling/expression-language)
as I wanted to reduce the amount of worrying about replacements and validation of numbers.
But I'm open to modifying this if you think it's worth doing!